promql/promqltest/test.go

```diff
@@ -708,14 +846,40 @@ func (ev *evalCmd) checkAnnotations(expr string, annos annotations.Annotations)
 	case ev.info && countInfo == 0:
 		return fmt.Errorf("expected info annotations evaluating query %q (line %d) but got none", expr, ev.line)
 	}
+
+	var (
+		warnings = make([]string, 0, countWarnings)
+		infos    = make([]string, 0, countInfo)
+	)
```

Comment:
I don't think we have to do this outside of a hot performance critical path. Let's just write

```suggestion
	var warnings, infos []string
```